pyscitt/pyscitt/receipt.py

```diff
@@ -21,27 +24,82 @@
+    7: "cti",
+}
+
+
+def display_cwt_key(item: Any) -> Union[int, str]:
+    """Convert a CWT key to a string for pretty-printing."""
+    out = str(item)
+    return COMMON_CWT_KEYS_MAP.get(item, out)
+
+
+def display_cbor_val(item: Any) -> str:
+    """Convert a CBOR item to a string for pretty-printing."""
+    out = str(item)
+    if hasattr(item, "__name__"):
+        out = item.__name__
+    elif isinstance(item, datetime.datetime):
+        out = item.isoformat()
+    elif type(item) is bytes:
+        out = item.hex()
+    return out
+
+
+def cbor_as_dict(cbor_obj: Any, cbor_obj_key: Any = None) -> Any:
     """
-    Return a representation of a list of COSE header parameters that
-    is amenable to pretty-printing.
+    Return a printable representation of a CBOR object.
     """
 
-    def display(item):
-        if hasattr(item, "__name__"):
-            return item.__name__
-        if type(item) is bytes:
-            return item.hex()
-        return item
+    # pycose will use class instances for known and registered headers instead of ints
+    if hasattr(cbor_obj_key, "identifier"):
+        if cbor_obj_key.identifier == crypto.SCITTReceipts.identifier:
+            parsed_receipts = []
+            for item in cbor_obj:
+                if type(item) is bytes:
+                    receipt_as_dict = Receipt.decode(item).as_dict()
+                else:
+                    receipt_as_dict = Receipt.from_cose_obj(item).as_dict()
+            parsed_receipts.append(receipt_as_dict)
+            return parsed_receipts
+        if cbor_obj_key.identifier == crypto.CWTClaims.identifier:
+            return {display_cwt_key(k): cbor_as_dict(v, k) for k, v in cbor_obj.items()}
+        if cbor_obj_key.identifier == X5chain.identifier:
+            return [base64.b64encode(cert).decode("ascii") for cert in cbor_obj]
+        if cbor_obj_key.identifier == KID.identifier:
+            return cbor_obj.decode()
+        if cbor_obj_key.identifier == X5t.identifier:
+            return {"alg": cbor_obj[0], "hash": cbor_obj[1].hex()}
+
+    if isinstance(cbor_obj, list):
+        if not cbor_obj_key:
+            cbor_obj_key = "idx"
+        out_key = display_cbor_val(cbor_obj_key)
+        return {
+            display_cbor_val(f"{out_key}_{idx}"): cbor_as_dict(v, f"{out_key}_{idx}")
+            for idx, v in enumerate(cbor_obj)
+        }
+
+    if isinstance(cbor_obj, dict):
+        return {display_cbor_val(k): cbor_as_dict(v, k) for k, v in cbor_obj.items()}
 
-    # Decode KID into a 'readable' text string if present.
-    hdr_dict = {display(k): display(v) for k, v in phdr.items()}
-    if hdr_dict.get("KID"):
-        hdr_dict["KID"] = bytes.fromhex(hdr_dict["KID"]).decode()
+    # attempt to decode nested cbor
+    if type(cbor_obj) is bytes:
+        try:
+            decoded = cbor2.loads(cbor_obj)
+            return cbor_as_dict(decoded, cbor_obj_key)
+        except (CBORError, UnicodeDecodeError):
+            pass
```

Comment:
It could make sense to at least print a log of the error/exception here, just for debugging purposes.

Reply:
Removed this block of coded and will let the bytes to just become a hex, there might be an edge case where it is necessary but not for the current usecases